models/ignore_and_fire.h

```diff
@@ -0,0 +1,325 @@
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef IGNORE_AND_FIRE_H
+#define IGNORE_AND_FIRE_H
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: neuron, integrate-and-fire, current-based
+
+Short description
++++++++++++++++++
+
+Neuron generating spikes at fixed intervals irrespective of inputs.
+
+Description
++++++++++++
+
+The ``ignore_and_fire`` neuron is a neuron model generating spikes with a fixed predefined inter-spike interval,
+irrespective of its inputs ("ignore"). In this simplest version of the ignore_and_fire neuron, the inputs from other
+neurons or devices are not processed at all (*). The ``ignore_and_fire`` neuron is primarily used for neuronal-network
+model validation and benchmarking purposes, in particular to evaluate the correctness and performance of connectivity
+generation and inter-neuron communication. It permits an easy scaling of the network size and/or connectivity without
+affecting the output spike statistics. The amount of network traffic is predefined by the user, and therefore fully
+controllable and predictable, irrespective of the network size and structure.
+
+(*) The model can easily be extended and equipped with any arbitrary input processing or (after-) spike generation
+dynamics to make it more similar and comparable to other non-ignorant neuron models. In such extended ignore_and_fire
+models, the spike emission process would still be decoupled from the intrinsic neuron dynamics.
+
+Parameters
+++++++++++
+
+The following parameters can be set in the status dictionary.
+
+================= ======= ======================================================
+ phase                     Phase
+ rate               1/s    Firing rate
+================= ======= ======================================================
+
+References
+++++++++++
+
+.. [1] Rotter S,  Diesmann M (1999). Exact simulation of
+       time-invariant linear systems with applications to neuronal
+       modeling. Biologial Cybernetics 81:381-402.
+       DOI: https://doi.org/10.1007/s004220050570
+.. [2] Diesmann M, Gewaltig M-O, Rotter S, & Aertsen A (2001). State
+       space analysis of synchronous spiking in cortical neural
+       networks. Neurocomputing 38-40:565-571.
+       DOI: https://doi.org/10.1016/S0925-2312(01)00409-X
+
+Sends
++++++
+
+SpikeEvent
+
+Receives
+++++++++
+
+SpikeEvent, CurrentEvent, DataLoggingRequest
+
+EndUserDocs */
+
+/**
```

Comment:
The following comment can be removed. There exists no propagator here and no matrix multiplications are needed.